### PR TITLE
memory: rename format_mem and format_swap to fromat and format_alt

### PR DESCRIPTION
**Pull request by madhur**
Fixes issue #1642

### CODE REVIEW

File: src/blocks/memory.rs [hunk 1]
```diff
@@ -44,7 +44,7 @@
 //! [[block]]
 //! block = "memory"
 //! format_mem = " $icon $mem_used_percents.eng(1) "
```
Comment (MaxVerevkin) [inline src/blocks/memory.rs:46]:
```suggestion
//! format = " $icon $mem_used_percents.eng(1) "
```

File: src/blocks/memory.rs [hunk 2]
```diff
@@ -139,17 +130,10 @@ pub async fn run(config: toml::Value, mut api: CommonApi) -> Result<()> {
...
-            Memtype::Swap => match swap_used / swap_total * 100. {
-                x if x > config.critical_swap => State::Critical,
-                x if x > config.warning_swap => State::Warning,
-                _ => State::Idle,
-            },
+        widget.state = match mem_used / mem_total * 100. {          
+                x if x > config.critical_mem || x > config.critical_swap => State::Critical,
+                x if x > config.warning_mem || x > config.warning_swap => State::Warning,
+                _ => State::Idle,           
         };
```
Comment (MaxVerevkin) [inline src/blocks/memory.rs:137]: `mem_used / mem_total` should be compared against `critical_mem` and `warning_mem`, while `swap_used / swap_total` should be compared against `critical_swap` and `warning` swap. If at lest one of them is "critical", then the block should have "critical" state. Same for "warning".

**Comment by MaxVerevkin** [comment]
`Memtype` type can be removed.

**Comment by MaxVerevkin** [comment]
Please run `rustfmt --edition 2021 src/blocks/memory.rs`. Apparently `cargo fmt` (and `cargo fmt --check`) doesn't work on blocks' files because their `mod <module>` is defined inside of a macro. This is a known limitation of `cargo fmt`: https://github.com/rust-lang/rustfmt/issues/4078

@ammgws I guess the CI should run something like `rustfmt --check --edition 2021 $(fd -e rs)` instead of `cargo fmt --check`?

File: src/blocks/memory.rs [hunk 3]
```diff
@@ -118,7 +110,7 @@ pub async fn run(config: toml::Value, mut api: CommonApi) -> Result<()> {
 
         widget.set_format(format.clone());
         widget.set_values(map! {
-            "icon" => Value::icon(api.get_icon(icon)?),
+            "icon" => Value::icon(api.get_icon("memory_mem")?),
```
Comment (MaxVerevkin) [inline src/blocks/memory.rs:113]: I think it might be a good idea to also add `$icon_swap`.

File: src/blocks/memory.rs [hunk 4]
```diff
@@ -86,18 +82,14 @@ pub async fn run(config: toml::Value, mut api: CommonApi) -> Result<()> {
...
-    )?;
-
-    let clickable = config.clickable;
-    let mut memtype = config.display_type;
-    let (mut icon, mut format) = match memtype {
-        Memtype::Memory => ("memory_mem", &format_mem),
-        Memtype::Swap => ("memory_swap", &format_swap),
+    let mut format_alt = match config.format_alt {
+        Some(f) => Some(f.with_default(
+            " $icon $swap_free.eng(3,B,M)/$swap_total.eng(3,B,M)($swap_used_percents.eng(2)) ",
```
Comment (MaxVerevkin) [inline src/blocks/memory.rs:90]: Usually `format_alt`s don't have a default value.

File: src/blocks/memory.rs [hunk 5]
```diff
@@ -139,19 +131,26 @@ pub async fn run(config: toml::Value, mut api: CommonApi) -> Result<()> {
...
+            _ => State::Idle,
+        };
+
+        if mem_state == State::Critical || swap_state == State::Critical {
+            widget.state = State::Critical;
+        } else if mem_state == State::Warning || swap_state == State::Warning {
+            widget.state = State::Warning;
+        } else {
+            widget.state = State::Idle;
+        }
```
Comment (MaxVerevkin) [inline src/blocks/memory.rs:152]:
```suggestion
        widget.state = if mem_state == State::Critical || swap_state == State::Critical {
            State::Critical
        } else if mem_state == State::Warning || swap_state == State::Warning {
            State::Warning
        } else {
            State::Idle
        };
```

File: src/blocks/memory.rs (hunk 3 above)
Comment (madhur) [reply src/blocks/memory.rs:109]: added `^icon_memory_swap`

File: NEWS.md [hunk 6]
```diff
@@ -85,6 +85,7 @@ This is a major release which rewrites the core code to be asynchronous.
 - **notmuch**: `name` option is removed and now you can use `format` to set the name
 - **{apt, dnf, pacman}**: `hide_when_uptodate` option is removed and now you can use `format_up_to_date = ""` to hide the block
 - **temperature**: `collapsed` option is removed and now you can use `format_alt = " $icon "` to maintain the behavior
+- **memory**: `clickable`, `format_mem` and `format_swap` are removed and now you can use `format` and `format_alt` to maintain the behavior
```
Comment (MaxVerevkin) [inline NEWS.md:88]:
```suggestion
- **memory**: `clickable`, `display_type`, `format_mem` and `format_swap` are removed and now you can use `format` and `format_alt` to maintain the behavior
```

File: src/blocks/memory.rs [hunk 7]
```diff
@@ -43,8 +41,8 @@
 //! ```toml
 //! [[block]]
 //! block = "memory"
-//! format_mem = " $icon $mem_used_percents.eng(1) "
-//! clickable = false
+//! format = " $icon $mem_used_percents.eng(1) "
+//! format_alt = " $icon $icon $swap_free.eng(3,B,M)/$swap_total.eng(3,B,M)($swap_used_percents.eng(2)) "
```
Comment (MaxVerevkin) [inline src/blocks/memory.rs:45]:
```suggestion
//! format_alt = " $icon_swap $swap_free.eng(3,B,M)/$swap_total.eng(3,B,M)($swap_used_percents.eng(2)) "
```

**Review by MaxVerevkin — APPROVED** [review]

**Pull request merged**